Libraries/LibWeb/CSS/StyleValues/EasingStyleValue.cpp

```diff
@@ -298,6 +298,16 @@ double EasingStyleValue::CubicBezier::evaluate_at(double input_progress, bool) c
             last_x = solution.x;
         }
 
+        // Ad-hoc: Sometimes the last entry in 'm_cached_x_samples' does not
+        // reach {x = 1}, which leads to an index out of bounds error when x is 1 and we
+        // try to linearly interpolate between nearby_index and nearby_index + 1
+        // (since nearby_index will point to the last entry when x is 1 and last entry has {x < 1})
+        // This fixes that by manually making sure the last entry is {x = 1} when x is 1
+        if (x == 1 && last_x < 1) {
```

Comment:
Why does this only apply when `x == 1.`? Isn't this always a problem when `x > last_x`?

Reply:
I can see it being a problem when `x > last_x` but `x == 1` is still a special case because the test case that I've added makes it so that `last_x == 0.9999999999999998` and `last_t == 1.0000000000000013` when the loop is done, which means if you do `last_t += 1. / 60.` one more time, you'll get a solution with `x == 0.9998351851851852` which is still a problem (since x is still > last_x).

Maybe this is a better approach?:
```C++
if (x > last_x) {
    if (x == 1) {
        auto solution = solve(1.);
        m_cached_x_samples.append(solution);
    } else {
        last_t += 1. / 60.;
        auto solution = solve(last_t);
        m_cached_x_samples.append(solution);
    }
}
```

Reply:
I've found a much simpler solution that I'll share with you; resolving this.